lib/sparql.jison

```diff
@@ -472,7 +477,21 @@ PrefixDecl
     ;
 SelectQuery
     : SelectClauseWildcard DatasetClause* WhereClause SolutionModifierNoGroup -> extend($1, groupDatasets($2), $3, $4)
-    | SelectClauseVars     DatasetClause* WhereClause SolutionModifier        -> extend($1, groupDatasets($2), $3, $4)
+    | SelectClauseVars     DatasetClause* WhereClause SolutionModifier 
+    {
+      // If there is a group clause, check if all variables that are selected are in the grouped clause
+      if ($4.group) {
+        for (let selectVar of $1.variables) {
+          if (!selectVar.expression) {
+            const selectVarId = getIdOfExpression(selectVar);
+            if ($4.group.filter(groupVar => getIdOfExpression(groupVar) === selectVarId).length === 0) {
```

Comment:
Instead of filtering and checking for length, we can just use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find (which would stop after the first match, and not create an array)